src/shared/install.c

```diff
@@ -889,8 +892,12 @@ static int find_symlinks_in_directory(
                                 free_and_replace(dest, x);
                         }
 
+                        q = path_extract_filename(dest, &fname);
+                        if (q < 0)
+                                return q;
+
                         /* Check if what the symlink points to matches what we are looking for */
-                        found_dest = streq(basename(dest), info->name);
+                        found_dest = streq(fname, info->name);
```

Comment:
Hmm, path_equal_filename?

Reply:
so i'd replace `streq` to `path_equal_filename`?